### PR TITLE
Added multi-speed fan use with 4Pipe Fancoil unit and water coils in Unitary System

### PR DESCRIPTION
**Pull request by Nigusse**
This branch adds multi-speed fan use with 4Pipe Fancoil unit and Unitary Systems.  In unitary system the multi-speed fan is to be used with chilled and hot water coils. In both units depending on the system load the fan cycles between speed or cycle on-off at the lowest fan speed. If the fan is continuously operating fan, then it runs at lowest speed continuously at low load or no load.

This branch is ready for review.  There is one example file that has failed due to transition requirement.

### CODE REVIEW

**Comment by Nigusse** [comment]
Only one example file "HVACTemplate-5ZoneUnitarySystem_new.idf" failed because of transition need in the new code. In the new code the object name "UnitarySystemPerformance:HeatPump:Multispeed" changed to "UnitarySystemPerformance:Multispeed".

**Comment by mjwitte** [comment]
@Nigusse You can make that change in expandobjects (just search/replace UnitarySystemPerformance:HeatPump:Multispeed --> UnitarySystemPerformance:Multispeed).  There should be only one or two occurences.  Or I can change it and push it into your branch.

**Comment by Nigusse** [comment]
@mjwitte if it is just updating the epfilter fortran source code, then I will do it. Thanks.

File: doc/src/docs/EngineeringReference/13c-EncyclopaedicRefs.md [hunk 1]
```diff
@@ -1,3 +1,21 @@
+<head>
```
Comment (Myoldmopar) [inline doc/src/docs/EngineeringReference/13c-EncyclopaedicRefs.md]: @Nigusse This snippet will have to be removed before merging in.

File: src/EnergyPlus/FanCoilUnits.cc [hunk 2]
```diff
@@ -2128,6 +2200,672 @@ namespace FanCoilUnits {
 	}
 
 	void
+		SimMultiStage4PipeFanCoil(
```
Comment (Myoldmopar) [inline src/EnergyPlus/FanCoilUnits.cc]: @Nigusse If you are making changes, could you make sure this function name and the parentheses/brackets are indented uniformly?

**Comment by Myoldmopar** [comment]
@Nigusse I made a few fairly minor comments in the PR above.  

In addition, there are some build warnings that need to be addressed.  These are mostly unused variables, which should be easy to clean up.  These might not be actually unused local variables, but rather unused `using` statements.  

In addition, the test file failure is because you have 
- In the CMakeLists.txt file, the file is named: FanCoilAutosize_MultiSpeedFan
- The filename is actually: FanCoilAutoSize_MultiSpeedFan

Note that the _S_ in autosize is capitalized in the filename.

**Comment by Nigusse** [comment]
@Myoldmopar I will address the comments and will push the updated code.

**Comment by Nigusse** [comment]
@Myoldmopar 
Addressed the following comments:
(1) removed the snippet from Eng Ref document
(2) removed unused variables
(3) aligned function name positions
(4) changed the file to FanCoilAutosize_MultiSpeedFan.idf

Made minor cleanup, merged develop to my branch and pushed the branch to remote.

**Comment by Myoldmopar** [comment]
Thanks @Nigusse, I'll perform a review of the source now that those initial issues are addressed.

File: testfiles/CMakeLists.txt [hunk 3]
```diff
@@ -201,6 +201,7 @@ ADD_SIMULATION_TEST(IDF_FILE ExhFiredAbsorptionChiller.idf EPW_FILE USA_IL_Chica
 ADD_SIMULATION_TEST(IDF_FILE ExteriorLightsAndEq.idf EPW_FILE USA_CO_Golden-NREL.724666_TMY3.epw)
 ADD_SIMULATION_TEST(IDF_FILE FanCoilAutoSize.idf EPW_FILE USA_CO_Golden-NREL.724666_TMY3.epw)
 ADD_SIMULATION_TEST(IDF_FILE FanCoilAutoSizeScalableSizing.idf EPW_FILE USA_IL_Chicago-OHare.Intl.AP.725300_TMY3.epw)
+ADD_SIMULATION_TEST(IDF_FILE FanCoilAutosize_MultiSpeedFan.idf EPW_FILE USA_IL_Chicago-OHare.Intl.AP.725300_TMY3.epw)
```
Comment (Myoldmopar) [inline testfiles/CMakeLists.txt]: Still shows a lower case s here

**Comment by Myoldmopar** [comment]
@Nigusse Great, it should be fine now.  

Could you also update the [initial comment](https://github.com/NREL/EnergyPlus/pull/5091#issue-100440369) on this PR with just a sentence or two describing the feature.  If a user clicks on the link for this feature in the changelog released with E+ that's the first thing they'll see.  Thanks!

**Comment by Nigusse** [comment]
@Myoldmopar added comment to the "initial comment".

**Comment by Myoldmopar** [comment]
@Nigusse I did a little cleanup in the docs and cleaned out an unnecessary member constructor in a header file.  The code, docs, unit tests, all look good.  If CI comes back clean, I am OK merging this in.

I'll give an FYI here to @mjwitte in case he wants to add further review, including the epfilter changes.

I'll also gave an FYI here to @EnergyArchmage.  I did a test and it appears this branch _won't_ conflict with his four pipe terminal branch.  But still...

File: idd/Energy+.idd.in [hunk 4]
```diff
@@ -31865,6 +31865,7 @@ ZoneHVAC:FourPipeFanCoil,
         \key CyclingFan
         \key VariableFanVariableFlow
         \key VariableFanConstantFlow
+        \key MultiStageFan
```
Comment (mjwitte) [inline idd/Energy+.idd.in:31868]: @Nigusse Curious why you chose "MultiStageFan" instead of "MultiSpeedFan"?

**Comment by Myoldmopar** [comment]
Yes, although very rarely.  The most prominent example is the [DataLoopNode](https://github.com/NREL/EnergyPlus/blob/f7698b73e319315ccba9c0859dc76bb5a58a801c/src/EnergyPlus/DataLoopNode.hh#L73) struct, with its large [member constructor](https://github.com/NREL/EnergyPlus/blob/f7698b73e319315ccba9c0859dc76bb5a58a801c/src/EnergyPlus/DataLoopNode.hh#L155).  It is used to initialize to a specific set of initial values [here](https://github.com/NREL/EnergyPlus/blob/f7698b73e319315ccba9c0859dc76bb5a58a801c/src/EnergyPlus/DataLoopNode.cc#L87), which is equivalent to [here](https://github.com/NREL/EnergyPlusArchive/blob/master/SourceCode/DataLoopNode.f90#L154).

**Comment by mjwitte** [comment]
@Nigusse @Myoldmopar epfilter changes look good.

**Comment by mjwitte** [comment]
@Myoldmopar But even there, those [initial values](https://github.com/NREL/EnergyPlusArchive/blob/master/SourceCode/DataLoopNode.f90#L154) appear to be identical to the [default constructor](https://github.com/NREL/EnergyPlus/blob/f7698b73e319315ccba9c0859dc76bb5a58a801c/src/EnergyPlus/DataLoopNode.hh#L115-L153) initial values, so that couldn't the member constructor and the list of [initial values](https://github.com/NREL/EnergyPlus/blob/f7698b73e319315ccba9c0859dc76bb5a58a801c/src/EnergyPlus/DataLoopNode.cc#L87) be eliminated?

**Comment by mjwitte** [comment]
@Myoldmopar Have you tested the new features and tried to break them?  I could do some of that tomorrow'ish.

File: idd/Energy+.idd.in (hunk 4 above)
Comment (Nigusse) [inline idd/Energy+.idd.in:31868]: @mjwitte No particular reason. I think, originally came from the user's request document.

Comment (mjwitte) [inline idd/Energy+.idd.in:31868]: @Nigusse That's interesting.  If that's a more common term in the field, the I won't argue.  Perhaps you could double-check with the original requestor.  Either way it needs explanation to distinguish it from the existing CyclingFan option.

Comment (Nigusse) [inline idd/Energy+.idd.in:31868]: @mjwitte Definitions are provided in the documentation. The MultiStageFan cycles between stages or cycles at the lowest fan speed.  The control is more of capacity staging by varying the fan speed. The CyclingFan is purely cycling at all speed unless the load is higher than the maximum speed.

**Comment by Myoldmopar** [comment]
@mjwitte That's fair on the node discussion.  We could remove it.  There may be others out there that _are_ used that I don't know about.  We could use a tool to check definitively, but I probably won't worry about that anytime soon.  I just prune them as I go.  

I haven't tried hard to break this.  If you are able to throw things at it, that would be awesome.

File: idd/Energy+.idd.in (hunk 4 above)
Comment (Nigusse) [inline idd/Energy+.idd.in:31868]: @mjwitte @Myoldmopar I contacted Jim Pegues of Carrier, he prefers the name "MultiSpeedFan" over "MultiStageFan".  So, the key name may have to change.

**Comment by Myoldmopar** [comment]
Merging.  @mjwitte, if you keep a list of things to try to break, add this to it and we'll open up an issue later.

**Pull request merged**